core/src/main/java/org/apache/iceberg/CatalogUtil.java

```diff
@@ -407,36 +408,38 @@ public static void configureHadoopConf(Object maybeConfigurable, Object conf) {
    *
    * <p>The implementation must have a no-arg constructor.
    *
-   * @param impl full class name of a custom {@link MetricsReporter} implementation
+   * @param properties catalog properties which contains class name of a custom {@link
+   *     MetricsReporter} implementation
    * @return An initialized {@link MetricsReporter}.
-   * @throws IllegalArgumentException if class path not found or right constructor not found or the
-   *     loaded class cannot be cast to the given interface type
    */
-  public static MetricsReporter loadMetricsReporter(String impl) {
+  public static MetricsReporter loadMetricsReporter(Map<String, String> properties) {
```

Comment:
I agree with everything what @rdblue mentioned here. 
* it should fallback to the `LoggingMetricsReporter` if there's no reporter available (so that me don't change behavior)
* failures should still happen (as before) if something is misconfigured

Reply:
PTAL